tests/json_test.go

```diff
@@ -0,0 +1,152 @@
+package tests
+
+import (
+	"testing"
+
+	"github.com/kinbiko/jsonassert"
+	"gotest.tools/v3/assert"
+)
+
+func TestJSONOperations(t *testing.T) {
+	conn := getLocalConnection()
+	defer conn.Close()
+	ja := jsonassert.New(t)
+
+	t.Run("Set and Get Integer", func(t *testing.T) {
+		setCmd := `JSON.SET tools $ 2`
+		result := fireCommand(conn, setCmd)
+		assert.Equal(t, "OK", result)
+
+		getCmd := `JSON.GET tools`
+		result = fireCommand(conn, getCmd)
+		ja.Assertf(result.(string), `2`)
+	})
+
+	t.Run("Set and Get Boolean True", func(t *testing.T) {
+		setCmd := `JSON.SET booleanTrue $ true`
+		result := fireCommand(conn, setCmd)
+		assert.Equal(t, "OK", result)
+
+		getCmd := `JSON.GET booleanTrue`
+		result = fireCommand(conn, getCmd)
+		ja.Assertf(result.(string), `true`)
+	})
+
+	t.Run("Set and Get Boolean False", func(t *testing.T) {
+		setCmd := `JSON.SET booleanFalse $ false`
+		result := fireCommand(conn, setCmd)
+		assert.Equal(t, "OK", result)
+
+		getCmd := `JSON.GET booleanFalse`
+		result = fireCommand(conn, getCmd)
+		ja.Assertf(result.(string), `false`)
+	})
+
+	t.Run("Set and Get Simple JSON", func(t *testing.T) {
+		setCmd := `JSON.SET user $ {"name":"John","age":30}`
+		result := fireCommand(conn, setCmd)
+		assert.Equal(t, "OK", result)
+
+		getCmd := `JSON.GET user`
+		result = fireCommand(conn, getCmd)
+
+		ja.Assertf(result.(string), `{"age":30,"name":"John"}`)
+	})
+
+	t.Run("Set and Get Nested JSON", func(t *testing.T) {
+		setCmd := `JSON.SET user:2 $ {"name":"Alice","address":{"city":"New York","zip":"10001"},"array":[1,2,3,4,5]}`
```

Comment:
It's better to store the JSON string representation being set and later used for assertion in a single variable and then use the variable in both places.
This improves maintainability and readability.